test/lit.cfg

```diff
@@ -311,6 +311,40 @@ if test_sdk_overlay_dir is not None:
 test_options = os.environ.get('SWIFT_TEST_OPTIONS')
 config.swift_test_options = ' ' + test_options if test_options else ''
 
+def get_clang_library_dir():
+    llvm_lib_clang = os.path.join(config.llvm_libs_dir, 'clang')
+    if not os.path.exists(llvm_lib_clang):
+        return ''
+
+    versions = os.listdir(llvm_lib_clang)
+    if not len(versions):
+        return ''
+
+    version = max(versions)
+    return os.path.join(llvm_lib_clang, version, 'lib',
+                        platform.system().lower())
+
+config.clang_library_dir = get_clang_library_dir()
```

Comment:
It looks like lib/swift/clang doesn't get a copy of libclang*. Should we change clang's cmake files to put the libraries under swift-macosx-$target/lib/ instead of llvm-macosx-$target?